File: internal/protocol/remote.go
package protocol

// RemoteProtocol 远程协议类型
type RemoteProtocol string

const (
	RemoteProtocolSSH  RemoteProtocol = "ssh"  // SSH
	RemoteProtocolRDP  RemoteProtocol = "rdp"  // RDP (Remote Desktop Protocol)
	RemoteProtocolVNC  RemoteProtocol = "vnc"  // VNC
	RemoteProtocolTelnet RemoteProtocol = "telnet" // Telnet
	RemoteProtocolFTP  RemoteProtocol = "ftp"  // FTP
)

// RemoteConnectionInfo 远程连接信息
type RemoteConnectionInfo struct {
	Protocol RemoteProtocol `json:"protocol"`     // 协议类型
	Host     string         `json:"host"`         // 目标主机
	Port     int            `json:"port"`         // 端口
	Username string         `json:"username"`     // 用户名（可选）
	Password string         `json:"password"`     // 密码（可选，通常不存储）
	AuthType string         `json:"authType"`     // 认证类型: password, key
	Name     string         `json:"name"`         // 连接名称
}

// RemoteServiceInfo 远程服务信息（能力检测返回）
type RemoteServiceInfo struct {
	Protocol RemoteProtocol `json:"protocol"`
	Host     string         `json:"host"`      // 监听地址 (0.0.0.0, 127.0.0.1)
	Port     int            `json:"port"`      // 监听端口
	Name     string         `json:"name"`      // 服务名称
	Running  bool           `json:"running"`   // 是否运行中
}

// RemoteProxyStartPayload 启动远程代理负载
type RemoteProxyStartPayload struct {
	ConnectionID string                `json:"connectionId"` // 连接ID
	Protocol     RemoteProtocol        `json:"protocol"`     // 协议类型
	Target       string                `json:"target"`        // 目标地址 host:port
	Timeout      int                   `json:"timeout"`      // 超时时间（秒）
}

// RemoteProxyDataPayload 远程代理数据负载
type RemoteProxyDataPayload struct {
	ConnectionID string `json:"connectionId"` // 连接ID
	Data         []byte `json:"data"`         // 数据
}

// RemoteProxyClosePayload 关闭远程代理负载
type RemoteProxyClosePayload struct {
	ConnectionID string `json:"connectionId"` // 连接ID
	Reason       string `json:"reason"`       // 关闭原因
}

File: web/src/services/remote.ts
import axios from 'axios'
import { logger } from '@/utils/logger'

export type RemoteProtocol = 'ssh' | 'rdp' | 'vnc' | 'telnet'

export interface RemoteTicketRequest {
  agentId: string
  host: string
  port: number
  protocol: RemoteProtocol
  username?: string
  password?: string
  domain?: string
  width?: number
  height?: number
}

export interface RemoteTicket {
  ticket: string
  expiresAt: string
}

export interface RemoteSession {
  id: string
  agentId: string
  userId: string
  username: string
  protocol: RemoteProtocol
  host: string
  port: number
  status: 'pending' | 'connected' | 'closed' | 'failed'
  error?: string
  createdAt: string
  updatedAt: string
  closedAt?: string
}

export interface CreateRemoteSessionRequest {
  agentId: string
  protocol: RemoteProtocol
  host: string
  port: number
}

const remoteClient = axios.create({
  baseURL: '/api/remote',
  timeout: 30000,
  headers: {
    'Content-Type': 'application/json',
  },
})

remoteClient.interceptors.request.use(
  (config) => {
    const token = localStorage.getItem('token')
    if (token) {
      config.headers.Authorization = `Bearer ${token}`
    }
    return config
  },
  (error) => Promise.reject(error)
)

remoteClient.interceptors.response.use(
  (response) => response.data,
  (error) => {
    if (error.response?.status === 401) {
      localStorage.removeItem('token')
      localStorage.removeItem('username')
      window.location.href = '/login'
    }
    logger.error('Remote API Error:', error)
    return Promise.reject(error)
  }
)

export async function createRemoteTicket(params: RemoteTicketRequest): Promise<RemoteTicket> {
  const response = await remoteClient.post<unknown, { ticket: string; expires_at: string }>('/tickets', {
    agent_id: params.agentId,
    host: params.host,
    port: params.port,
    protocol: params.protocol,
    username: params.username,
    password: params.password,
    domain: params.domain,
    width: params.width,
    height: params.height,
  })

  return {
    ticket: response.ticket,
    expiresAt: response.expires_at,
  }
}

export async function getRemoteSessions(): Promise<RemoteSession[]> {
  const response = await remoteClient.get<unknown, { data: RemoteSession[] }>('/sessions')
  return response.data
}

export async function createRemoteSession(data: CreateRemoteSessionRequest): Promise<RemoteSession> {
  return remoteClient.post<unknown, RemoteSession>('/sessions', data)
}

export async function getRemoteSession(id: string): Promise<RemoteSession> {
  return remoteClient.get<unknown, RemoteSession>(`/sessions/${id}`)
}

export async function deleteRemoteSession(id: string): Promise<void> {
  return remoteClient.delete(`/sessions/${id}`)
}
